request/router.py

```diff
@@ -18,7 +18,7 @@ def __init__(self, unknown, *args):
         self.unknown = unknown
 
         for pattern in args:
-            if pattern.__class__ == str:
+            if pattern.__class__ in (str, u''.__class__):
```

Comment:
You should test it with ``isinstance(pattern, six.string_types)`` (from ``django.utils.six``).